src/lib/support/PersistedCounter.h

```diff
@@ -130,9 +169,15 @@ class PersistedCounter : public MonotonicallyIncreasingCounter<T>
             // Ensure that a new starting point is persisted.
             ReturnErrorOnFailure(PersistNextEpochStart(mNextEpoch + mEpoch));
 
-            // Advancing the epoch should have ensured that the current value
-            // is valid
-            VerifyOrReturnError(MonotonicallyIncreasingCounter<T>::GetValue() < mNextEpoch, CHIP_ERROR_INTERNAL);
+            // Advancing the epoch should have ensured that the current value is valid
+            VerifyOrReturnError(MonotonicallyIncreasingCounter<T>::GetValue() + mEpoch == mNextEpoch, CHIP_ERROR_INTERNAL);
```

Comment:
It's sort of odd to fix this bit but not the `GetValue() >= mNextEpoch` check... Or is 33175 tracking fixing this in general?

Reply:
Reason for the difference in `>=` and `==` was due to this bit of logic
```
// Calculate previous Epoch from value
T previousEpoch = value - (value % mEpoch);
```

which created a situation where it was possible for `==` would be false. By simplifying the logic around that calculation, both can have the same check.